test/distributed/org/apache/cassandra/distributed/test/AuthTest.java

```diff
@@ -52,4 +64,60 @@ public void authSetupIsCalledAfterStartup() throws IOException
             assertTrue(setupCalled);
         }
     }
+
+    /**
+     * Sows that in some circumstances CassandraRoleManager will create the cassandra role twice
+     */
+    @Test
+    public void authSetupIsCalledTwice() throws IOException
+    {
+        try (Cluster cluster = builder().withNodes(1)
+                                        .withTokenSupplier(TokenSupplier.evenlyDistributedTokens(3))
+                                        .withNodeIdTopology(NetworkTopology.singleDcNetworkTopology(3, "dc0", "rack0"))
+                                        .withConfig(config -> config.with(NETWORK, GOSSIP).set("authenticator", "PasswordAuthenticator"))
+                                        .start())
+        {
+            boolean defaultRoleSetup = cluster.get(1).callOnInstance(() -> {
```

Comment:
This whole logic can be replaced by Awaitility library. We would just poll for 10s to get true on hasExistingRoles and it would hide all lowlevel complexity here.